File: src/shared/protocol.zig
pub const ServerMessage = union(enum) {
    data: []const u8,
    ack: Size,
};

pub const ClientMessage = union(enum) {
    input: []const u8,
    resize: Size,
};

pub const Size = struct {
    cols: u16,
    rows: u16,
};

File: client/src/protocol.rs
//! Wire-protocol envelopes. Mirror of `server/src/protocol.rs`.

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize)]
#[serde(tag = "type", rename_all = "lowercase")]
pub enum ClientMsg {
    Input { value: String },
    Resize { cols: u16, rows: u16 },
}

#[derive(Debug, Clone, Deserialize)]
#[serde(tag = "type", rename_all = "lowercase")]
pub enum ServerMsg {
    Data { value: String },
    Ack { cols: u16, rows: u16 },
}
